tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/EmailAddressContactFactory.java

```diff
@@ -0,0 +1,180 @@
+        }
+
+        private void createIndexIfNeeded(ReactorElasticSearchClient client, IndexName indexName, XContentBuilder settings, Optional<XContentBuilder> mappingContent)
+        throws IOException {
+            try {
+                if (!indexExists(client, indexName)) {
+                    CreateIndexRequest request = new CreateIndexRequest(indexName.getValue()).source(settings);
+                    mappingContent.ifPresent(request::mapping);
+                    client.indices().create(
+                            request,
+                            RequestOptions.DEFAULT);
+                }
+            } catch (ElasticsearchStatusException exception) {
+                if (exception.getMessage().contains(INDEX_ALREADY_EXISTS_EXCEPTION_MESSAGE));
+            }
+        }
+
+        private boolean indexExists(ReactorElasticSearchClient client, IndexName indexName) throws IOException {
+            return client.indices().exists(new GetIndexRequest(indexName.getValue()), RequestOptions.DEFAULT);
+        }
+
+        private XContentBuilder generateSetting(int nbShards, int nbReplica, int waitForActiveShards) throws IOException {
+            return jsonBuilder()
+                    .startObject()
+                        .startObject("settings")
+                            .field("number_of_shards", nbShards)
+                            .field("number_of_replicas", nbReplica)
+                            .field("index.write.wait_for_active_shards", waitForActiveShards)
+                            .startObject("mappings")
+                                .startObject("properties")
+                                    .startObject("accountId")
+                                        .field("type", "text")
+                                    .endObject()
+                                    .startObject("id")
+                                        .field("type", "text")
+                                    .endObject()
+                                    .startObject("address")
+                                        .field("type", "text")
+                                    .endObject()
+                                .endObject()
+                            .endObject()
+                            .startObject("analysis")
+                                .startObject("analyzer")
+                                    .startObject("contact_analyzer")
+                                    .field("tokenizer", "contact_tokenizer")
+                                    .endObject()
+                                .endObject()
+                            .endObject()
+                            .startObject("tokenizer")
+                                .startObject("contact_tokenizer")
+                                .field("type", "ngram")
+                                .field("min_gram", 2)
+                                .field("max_gram", 20)
```

Comment:
```suggestion
                                .field("min_gram", 3)
                                .field("max_gram", 3)
```

tri-grams should be enough for accuracy. using more ngrams comes at an expensive computing cost!

Reply:
Reasonable, fixed!